src/acachecontrol/request_context_manager.py

```diff
@@ -14,29 +14,32 @@ def __init__(self, client_session, cache, method, url, **params):
     async def __aenter__(self):
         await self.cache.register_new_key(self.cache_key)
 
-        if self.cache_key not in self.cache:
+        if not self.cache.has_valid_entry(self.cache_key):
```

Comment:
`self.cache_key` confuse me. I'd rename it to `self.key` since we use `cache_key` in `AsyncCache` for hashable key whereas `self.cache_key` is not hashable.